stablehlo/dialect/Base.cpp

```diff
@@ -381,6 +381,52 @@ FailureOr<Type> inferMostSpecificType(std::optional<Location> location,
                                inferMostSpecificDimAndBound);
 }
 
+FailureOr<Type> mapOverTupleElements(
+    std::optional<Location> location, TypeRange inputTypes,
+    function_ref<FailureOr<Type>(std::optional<Location>, TypeRange types)>
+        fn) {
+  SmallVector<TupleType> tupleTypes;
+  for (auto inputType : inputTypes) {
+    if (auto tupleType = inputType.dyn_cast<TupleType>())
+      tupleTypes.push_back(tupleType);
+  }
+  if (!tupleTypes.empty()) {
+    if (tupleTypes.size() != inputTypes.size())
+      return emitOptionalError(location,
+                               "Mismatched type kinds: either all types ",
+                               "must be tuples, or no types must be tuples");
+    SmallVector<Type> results(tupleTypes[0].size());
+    for (auto tupleType : tupleTypes) {
+      if (tupleType.size() != results.size())
+        return emitOptionalError(location,
```

Comment:
similar question, surprised by this restriction, would be good to add to comment at top of function

Reply:
Done.